pymc3/sampling.py

```diff
@@ -412,6 +419,30 @@ def sample(
     if not isinstance(random_seed, Iterable):
         raise TypeError("Invalid value for `random_seed`. Must be tuple, list or int")
 
+    if not discard_tuned_samples and not return_inferencedata:
+        warnings.warn(
+            "Tuning samples will be included in the returned `MultiTrace` object. "
+            "To avoid complications with downstream analysis, you should slice them out:\n"
+            "trace_warmup = trace[: -trace.report.n_draws]\n"
+            "trace_posterior = trace[-trace.report.n_draws :]\n"
+            "Alternatively, you can use InferenceData: `pm.sample(..., return_inferencedata=True)`",
+            UserWarning
+        )
+
+    if return_inferencedata is None:
+        warnings.warn(
+            "In v4.0.0, pm.sample will return an `arviz.InferenceData` object instead of a `MultiTrace` by default. "
+            "You can pass return_inferencedata=True or return_inferencedata=False to be safe and silence this warning.",
+            FutureWarning
+        )
+        # set the default
+        return_inferencedata = False
+    if return_inferencedata and not discard_tuned_samples:
+        raise NotImplementedError(
+            'arviz.from_pymc3 does not handle warmups yet. See ArviZ issue #1146.'
+            'For this reason, we can not return InferenceData that includes warmup draws right now.'
+        )
```

Comment:
I think this warning should be deleted, as the ArviZ PR will be merged in the next few days -- and will definitely be merged when this goes out with 3.9

Reply:
I will take this out as soon as the ArviZ release number is known - then we can pin on that, or check the version here to maintain compatibility with ArviZ 0.7.0, if anything speaks against pinning `arviz>=0.?.?`

Reply:
I'd be in favor of pinning latest ArviZ, as this would simplify the code and workflow, and 0.x versions aren't supposed to be "production stable". But I'm of course open to debate.
I'll let know as soon as the ArviZ number is known

Reply:
Done!